addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -87,13 +91,61 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", scr.Name, scr.Namespace, err)
 	}
 
+	err = r.createS3(ctx, req, mirrorPeer)
+	if err != nil {
+		klog.Error(err, "Failed to create ODR S3 resources")
+		return ctrl.Result{}, err
+	}
+
 	err = r.createVolumeReplicationClass(ctx, &mirrorPeer)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to create volumereplicationclass: %v", err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *MirrorPeerReconciler) createS3(ctx context.Context, req ctrl.Request, mirrorPeer multiclusterv1alpha1.MirrorPeer) error {
+	var err error
+
+	var peerAccumulator string
+	for _, peer := range mirrorPeer.Spec.Items {
+		peerAccumulator += peer.ClusterName
+	}
+	checksum := sha1.Sum([]byte(peerAccumulator))
+
+	bucketGenerateName := "odrsystem-"
```

Comment:
it is not  a good practice to declare constant inbetween, also create contants without "-" and format it inside core logic